src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerSessionMonitor.cs

```diff
@@ -186,113 +185,88 @@ private bool RemoveSession(int sessionId, out ProfilerSession session)
             }
-        }
-
-        private List<ProfilerEvent> PollSession(ProfilerSession session)
-        {
-            var events = new List<ProfilerEvent>();
-            try
+            foreach (var kvp in xEvent.Actions)
             {
-                if (session == null || session.XEventSession == null)
-                {
-                    return events;
-                }
-
-                var targetXml = session.XEventSession.GetTargetXml();
+                profileEvent.Values.Add(kvp.Key, kvp.Value.ToString());
+            }
+            var eventList = new List<ProfilerEvent>();
+            eventList.Add(profileEvent);
+            var eventsLost = session.EventsLost;
 
-                XmlDocument xmlDoc = new XmlDocument();
-                xmlDoc.LoadXml(targetXml);
+            if (eventList.Count > 0 || eventsLost)
+            {
+                session.FilterOldEvents(eventList);
+                eventList = session.FilterProfilerEvents(eventList);
+                // notify all viewers of the event.
+                List<string> viewerIds = this.sessionViewers[session.XEventSession.Id];
 
-                var nodes = xmlDoc.DocumentElement.GetElementsByTagName("event");
-                foreach (XmlNode node in nodes)
+                foreach (string viewerId in viewerIds)
                 {
-                    var profilerEvent = ParseProfilerEvent(node);
-                    if (profilerEvent != null)
+                    if (allViewers[viewerId].active)
                     {
-                        events.Add(profilerEvent);
+                        SendEventsToListeners(viewerId, eventList, eventsLost);
                     }
                 }
             }
-            catch (XEventException)
-            {
-                SendStoppedSessionInfoToListeners(session.XEventSession.Id);
-                ProfilerSession tempSession;
-                RemoveSession(session.XEventSession.Id, out tempSession);
-            }
-            catch (Exception ex)
-            {
-                Logger.Write(TraceEventType.Warning, "Failed to poll session. error: " + ex.Message);
-            }
-            finally
+        }
+
+        /// <summary>
+        /// Function that creates a brand new stream from a session, this is called from ProcessSessions when a session doesn't have a stream running currently.
+        /// </summary>
+        private void StartStream(int id, ProfilerSession session)
+        {
+            CancellationTokenSource threadCancellationToken = new CancellationTokenSource();
+            var connectionString = ConnectionService.BuildConnectionString((session.XEventSession as XEventSession).ConnDetails);
+            var eventStreamer = new XELiveEventStreamer(connectionString, (session.XEventSession as XEventSession).Session?.Name);
+            //Start streaming task here, will run until cancellation or error with the feed.
```

Comment:
nit, for all the code comments, there should be a space between // and the comment.